### PR TITLE
fix: session view page error on clicking join event

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/6823

https://github.com/fossasia/open-event-frontend/pull/6795

https://user-images.githubusercontent.com/56407566/109376043-cb636780-78e7-11eb-9914-3c9dbae8964d.mp4

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/public/stream/join-video.ts [hunk 1]
```diff
@@ -36,11 +36,15 @@ export default class JoinVideo extends Component<Args> {
 
   @action
   openPanel(): void {
+    let currentRoute  = this.router.currentRoute.name;
+    if (currentRoute === 'public.session.view' || currentRoute === 'public.exhibition.view') {
+      currentRoute = 'public';
```
Comment (iamareebjamal) [inline app/components/public/stream/join-video.ts:41]: Why?

Comment (maze-runnar) [reply app/components/public/stream/join-video.ts:41]: on the session view page it needs session-Id also to transition.

Comment (iamareebjamal) [reply app/components/public/stream/join-video.ts:41]: There must be a better solution

**Comment by maze-runnar** [comment]
https://user-images.githubusercontent.com/56407566/109525634-ba794880-7ad7-11eb-95ed-785876f9e672.mp4

**Comment by mariobehling** [comment]
Does it work if you click on "Join Video" on the schedule and the session page itself - in the green area in the below screenshot?

Screenshot from 2021-03-01 19-41-08

**Comment by iamareebjamal** [comment]
Instead, pass in a callback in args which supply their own route instead of parsing the current route. It doesn't belong in the component and is brittle

**Comment by maze-runnar** [comment]
I couldn't find another way.

**Comment by iamareebjamal** [comment]
@maze-runnar See the solution

**Pull request merged**